File: JM/Hash/3. Phone_book.java
//전화번호 목록
import java.util.*;

class Solution {
    public boolean solution(String[] phone_book) {
        Map<String, Integer> pmap = new HashMap<String, Integer>();
        for(String phone_num : phone_book){
            pmap.put(phone_num, 1);
        }
        
        boolean answer = true;
        for(String phone_num : phone_book){
            for(int i=1;i<phone_num.length();i++){
                String temp_pnum = phone_num.substring(0, i);
                if(pmap.containsKey(temp_pnum)) return false;
            }
        }
        return answer;
    }
}

File: YJ/01_Hash/3_phonebook.py
def solution(phone_book):
    length = len(phone_book)
    
    for i in range(length):
        for j in range(length):
            if i != j and phone_book[j].startswith(phone_book[i]): #접두 전화번호 비교
                return False
    
    return True